infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/WaitSnapshotApplyState.java

```diff
@@ -117,6 +120,7 @@ public LogReplicationState processEvent(LogReplicationEvent event) throws Illega
                 return this;
             case REPLICATION_STOP:
                 log.debug("Stop Log Replication while waiting for snapshot sync apply to complete id={}", transitionEventId);
+                stopSnapshotApply.set(true);
```

Comment:
Should we update this log message? looks like we don't expect SNAPSHOT_APPLY_IN_PROGRESS when stopSnapshotAppy is set to true?

Reply:
oh no this just mean it's stopping log replication while in X state.